pkg/pod-autoscaler/pkg/pod-resource-updater/controller.go

```diff
@@ -134,43 +134,66 @@ func (c *Controller) runNodeScaleWorker() {
 				return
 			}
 
+			isReady := false
+			for _, c := range pod.Status.Conditions {
+				if c.Type == corev1.PodReady && c.Status == corev1.ConditionTrue {
+					isReady = true
+				}
+			}
+
+			if !isReady {
+				klog.Infof("Pod %s/%s not ready", pod.Namespace, pod.Name)
+				return
+			}
+
 			newPod, err := syncPod(pod, *podScale)
 			if err != nil {
 				klog.Error("Error syncing the pod: ", err)
 				return
 			}
 
-			// try both updates in dry-run first and then actuate them consistently
-			updatedPod, updatedPodScale, err := c.AtomicResourceUpdate(newPod, podScale)
-
-			if err != nil {
-				klog.Error("Error while updating pod and podscale: ", err)
-				//TODO: We are using this channel as a workqueue. Why don't use one?
-				c.in <- nodeScale
-				return
+			requireUpdate := false
+			for i := range pod.Spec.Containers {
+				if pod.Spec.Containers[i].Resources.Requests.Cpu().MilliValue() !=
+					newPod.Spec.Containers[i].Resources.Requests.Cpu().MilliValue() {
+					requireUpdate = true
+				}
 			}
 
-			//TODO: handle error
-			_ = c.log.Log(updatedPodScale)
+			if requireUpdate {
```

Comment:
To avoid indentation can you change this code block into:
```
if !requiredUpdate{
    return
}

.... remaining function body
```

Reply:
solved